j5/backends/base.py

```diff
@@ -51,7 +51,7 @@ class Backend(metaclass=BackendMeta):
 
     @property
     @abstractmethod
-    def environment(self):
+    def environment(self) -> 'Environment':
```

Comment:
Why do quotes need to be used here?

Reply:
Because `Environment` hasn't yet been declared when it gets to this line in the file, causing Python to fall over. MyPy will see the strings and understand that it means `Environment` as they look for all classes and types before checking validity